pyo3-macros-backend/src/pyclass.rs

```diff
@@ -163,9 +168,17 @@ impl PyClassPyO3Options {
             };
         }
 
+        let python_version = pyo3_build_config::get().version;
+
         match option {
             PyClassPyO3Option::Crate(krate) => set_option!(krate),
-            PyClassPyO3Option::Dict(dict) => set_option!(dict),
+            PyClassPyO3Option::Dict(dict) => {
+                ensure_spanned!(
+                    python_version >= PY_3_9 || !is_abi3(),
+                    dict.span() => "`dict` requires >= python 3.9 with the `abi3` feature"
```

Comment:
```suggestion
                    dict.span() => "`dict` requires Python >= 3.9 with the `abi3` feature"
```
I think this is a bit easier to parse if this is ordered like an equation would, or alternatively phrased in words like "... requires a Python version greater than 3.9 ..."

(Same for the other one)